server/src/test/java/org/apache/druid/segment/realtime/appenderator/BatchAppenderatorTest.java

```diff
@@ -382,12 +387,14 @@ public void testTaskDoesNotFailAsExceededMemoryWithSkipBytesInMemoryOverheadChec
 
       appenderator.startJob();
       appenderator.add(IDENTIFIERS.get(0), createInputRow("2000", "foo", 1), null);
+      waitForPersists();
```

Comment:
I think you could keep the future of persist triggered in `add()` in a variable. Then you can add a method used only for testing that returns the persist future. Then you can finally wait for the future to be done instead of sleeping.

Reply:
I was trying to avoid more changes to the class (exposing methods etc) but I can pursue your suggestion